### PR TITLE
PICARD-740: Added missing album and track match quality icon info

### PR DESCRIPTION
**Pull request by suhas2go**
With reference to https://tickets.metabrainz.org/browse/PICARD-740, the issue here is that users are unable to understand the meaning of the different album icons that Picard uses. 

Since pyqt4 tooltips are not cross-platform friendly(refer https://bugreports.qt.io/browse/QTBUG-26669).In this implementation, I have displayed messages on the status bar to indicate the Album information. 

Although the original issue only deals with optical CD icons, I noticed that track match quality info is also missing. I have indicated the match quality using descriptors ranging from 'bad' to 'excellent'. This is slightly different from that in the documentation (https://picard.musicbrainz.org/docs/#matching-files) as it is difficult to find 6 levels of descriptors between 'good' and 'bad'.

Screenshots
album_info
track_info

### CODE REVIEW

**Comment by suhas2go** [comment]
I noticed a few bugs on Qt associated with tooltips, they seem to interfere with GUI testing and are highly unreliable on Mac OS. Although the above referred Qt issue is closed, I wanted to implement a safer solution. Therefore, I chose to display the information on the status bar. Also, I thought the information text is itself a little long for a tooltip.

**Comment by mineo** [comment]
It looks like those Qt bugs only apply to Qt 5, which we don't support, so maybe we can actually use tooltips here?

**Comment by suhas2go** [comment]
The only thing that was stopping me was Zas' comment on the issue but I talked to him. I've changed the status tips to tooltips and reduced text length to suit that of tooltips.

File: picard/ui/itemviews.py [hunk 1]
```diff
@@ -689,16 +698,21 @@ def update(self, update_tracks=True):
                     item.update(update_album=False)
         if album.errors:
             self.setIcon(0, AlbumItem.icon_error)
+            self.setToolTip(0,"Error")
```
Comment (zas) [inline picard/ui/itemviews.py]: Please add a space after the comma (PEP8)
Use `N_()` and `_()` to mark strings for translation where applicable.

File: picard/ui/itemviews.py [hunk 2]
```diff
@@ -689,16 +698,21 @@ def update(self, update_tracks=True):
...
         elif album.is_complete():
             if album.is_modified():
                 self.setIcon(0, AlbumItem.icon_cd_saved_modified)
+                self.setToolTip(0,"Album modified and complete")
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 3]
```diff
@@ -689,16 +698,21 @@ def update(self, update_tracks=True):
...
             else:
                 self.setIcon(0, AlbumItem.icon_cd_saved)
+                self.setToolTip(0,"Album unchanged and complete")
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 4]
```diff
@@ -689,16 +698,21 @@ def update(self, update_tracks=True):
...
         else:
             if album.is_modified():
                 self.setIcon(0, AlbumItem.icon_cd_modified)
+                self.setToolTip(0,"Album modified")
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 5]
```diff
@@ -689,16 +698,21 @@ def update(self, update_tracks=True):
...
             else:
                 self.setIcon(0, AlbumItem.icon_cd)
+                self.setToolTip(0,"Album unchanged")
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 6]
```diff
@@ -788,3 +803,19 @@ def decide_file_icon(file):
             return FileItem.icon_file_pending
         else:
             return FileItem.icon_file
+
+    @staticmethod
+    def decide_file_icon_info(file):
+        # Note error state info is already handled
+        if isinstance(file.parent, Track):
+            if file.state == File.NORMAL:
+                return "Track saved"
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 7]
```diff
@@ -788,3 +803,19 @@ def decide_file_icon(file):
...
+            elif file.state == File.PENDING:   # unsure how to use int(file.similarity * 5 + 0.5)
+                return "Pending "
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n, **extra space at end of string**

File: picard/ui/itemviews.py [hunk 8]
```diff
@@ -158,6 +158,14 @@ def create_icons(self):
...
             QtGui.QIcon(":/images/match-100.png"),
         ]
+        FileItem.match_icons_info = [
+            "Bad match",
+            "Poor match",
+            "Ok match",
+            "Good match",
+            "Great match",
+            "Excellent match",
+        ]
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

File: picard/ui/itemviews.py [hunk 9]
```diff
@@ -788,3 +803,19 @@ def decide_file_icon(file):
...
+        # Note error state info is already handled
+        if isinstance(file.parent, Track):
+            if file.state == File.NORMAL:
+                return "Track saved"
+            elif file.state == File.PENDING:   # unsure how to use int(file.similarity * 5 + 0.5)
+                return "Pending "
+            else:   # returns description of the match ranging from bad to excellent
+                return FileItem.match_icons_info[int(file.similarity * 5 + 0.5)]
+        elif file.state == File.PENDING:
+            return "Pending"
```
Comment (zas) [inline picard/ui/itemviews.py]: i18n

**Comment by suhas2go** [comment]
I've marked the strings for translation. I asked on the IRC but didn't get any response - Doesn't this mean I'll have to regenerate the message catalogs or is this done at a later stage?

**Comment by mineo** [comment]
You can regenerate the message catalogs, but it's not required. We definitely try to regenerate them after merging pull requests and/or sometime before a new release to give translators enough time to translate new strings.

**Pull request merged**